### PR TITLE
Add compatibility for new dump

### PR DESCRIPTION
**Pull request by ben-chain**
## Description
https://github.com/ethereum-optimism/contracts/pull/254 will lead to a new storage layout for `OVM_ETH` which needs to be reflected in geth.  This PR adds a conditional branch to deal with either case.

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](./CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: core/genesis.go [hunk 1]
```diff
@@ -300,11 +300,20 @@ func ApplyOvmStateToState(statedb *state.StateDB, stateDump *dump.OvmDump, l1XDo
 	}
 	OVM_ETH, ok := stateDump.Accounts["OVM_ETH"]
 	if ok {
-		// Set the gateway of OVM_ETH
 		log.Info("Setting OVM_L1ETHGateway in OVM_ETH", "address", l1ETHGatewayAddress.Hex())
-		l1GatewaySlot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000008")
-		l1GatewayValue := common.BytesToHash(l1ETHGatewayAddress.Bytes())
-		statedb.SetState(OVM_ETH.Address, l1GatewaySlot, l1GatewayValue)
+		if strings.Contains(OVM_ETH.Code, "a84ce98") {
```
Comment (ben-chain) [inline core/genesis.go:304]: this is a methodID not present in the legacy contract

**Comment by tynes** [comment]
Its not scalable to need to modify the storage slot directly longer term. Maybe we can migrate to using something like `core.ApplyTransaction` to do this? I'm ok with doing this approach for now

File: core/genesis.go (hunk 1 above)
Comment (smartcontracts) [reply core/genesis.go:304]: Fine for now.

**Review by smartcontracts — APPROVED** [review]
LGTM, let's revert this as soon as we can.

**Pull request merged**